### ISSUE TITLE
Logic-bound units stuck surrounded by other units prioritizing player "move" command

### ISSUE THREAD

**Issue by Lamby777** [Anuken/Mindustry#7692]
[VIDEO OF THE BUG](https://youtu.be/oiAK3MAvN2o) (used my friend's logic script cuz I was too lazy to copy over my own, but it's essentially the same besides that last part where the Eclipses get stuck trying to fly back over to the processor once finished loading up on blast. The units getting stuck at the core is normal gameplay, but the **units taking forever to actually bind to the processor and start heading over to the core** is the bug I'm reporting here)

**Platform**: W10 on Steam

**Build**: Discovered on 138, tested on 139 as well

**Issue**: If you tell a bunch of Zeniths to go over to <insert random point here> and then you turn on a logic processor that binds to them, only the few Zeniths that have reached that exact point will start binding to the processor, but they'll be stuck in the middle, surrounded by all the others that haven't completed their path yet. This makes the processor's code run pretty poorly because it runs for batches of 1-4 lucky units that made it to the middle and escaped the giant cluster of drones rather than for all 24 of them together.

**Steps to reproduce**: Explained above (tested with Zeniths and Eclipses, not sure if other units work too, but I'd assume any flying unit would do. Maybe even ground units?) ^^^

**Link(s) to mod(s) used**: [Foo's Client](https://github.com/mindustry-antigrief/mindustry-client), version 2d7cd75 9/29/2022 Unstable

**Save file**: [w.zip](https://github.com/Anuken/Mindustry/files/9750084/w.zip)
_(pls don't reap my soul uwu)_

**(Crash) logs**: amogus in real life sus sus playin amogus in real life spaceship with my crew gotta split up spread out cuz we got tasks to do

---

*Place an X (no spaces) between the brackets to confirm that you have read the line below.*  
- [X] **I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.**

- [X] **I have searched the closed and open issues to make sure that this problem has not already been reported.** (It might have been reported already, but I searched pretty hard. Most of the reports were about units getting stuck on doors and that sort of stuff.)

**Comment by Anuken** [comment]
Will not even be debugging this unless you provide the **exact** instruction and command that is malfunctioning, the expected results, and the actual results. The error reported here is very vague (tell? with what? go over where? with which instruction? binding to what processor?) and *I do not debug logic programs in general.*

**Comment by Lamby777** [comment]
Just edited the issue description, it looked weird because part of the message got hidden due to being inside angle brackets. The problem isn't with a logic program, rather with the way the AI handles logic processors attempting to bind to them while the AI is busy trying to reach a certain point given by the player's "move" command with shift + right click. In the video, you can see what I mean because as soon as I start shift right clicking on each individual unit, they unbind, because they're close enough to the pathfinding destination to switch over to their idle AI and then accept the logic's attempt to bind to it.

TL;DR Units following a player's "move" command will refuse to be bound to a logic processor until they reach their destination, and this means the ones that reach the destination will get stuck in between all the others still trying to reach it.

**Comment by Anuken** [comment]
I won't be changing that behavior. There is absolutely no grouping implemented for player-controlled units like that, so until they actually reach their destination - and not "their group" does, their task will not be considered complete.

**Comment by Lamby777** [comment]
Shouldn't the logic processor's bind instruction override any currently running move instruction to begin with, though? Changing that would fix this issue without any sort of "group" action system needing to be implemented, and seems like the far easier solution. I'm kinda bad at Java, but I could give it a try.

**Comment by Anuken** [comment]
No, it shouldn't. That is intentional. Logic being able to take over any player-commanded units would be a disaster, especially considering there's no good way to get the units back.

**Comment by Lamby777** [comment]
Okay, after thinking about this for a little, I think I have a solution that won't break anything and is also pretty easy to implement + learn how to use.

The same way all the support air units have stuff like Mine, Repair, Rebuild, etc. in the command menu, there could be a new universal command for every unit, including attack units. What this command would do is pretty much just force the unit into its idle mode, aborting all activities and getting rid of any movement plans it's currently following.

For this specific purpose, you could tell your Zenith drones to discard their paths, and that way all 24 would immediately be available for the processor, instead of each one taking turns reaching the destination and then being free.

If the user specifically tells the game to do whatever with the drone, they clearly aren't in the middle of an attack or anything and won't mind if a logic processor takes over.
